rules/prevent-abbreviations.js

```diff
@@ -231,146 +217,105 @@ const prepareOptions = ({
-
-This is fine since result of the function is used only to check if there is zero, one or more replacements and when formating the message.
-
-Example: `[[1, 2], [3, 4]]` → `[[1, 3], [1, 4], [2, 3], [2, 4]]`
-*/
-const getWordByWordReplacementsCombinations = (wordByWordReplacements, limit = 16) => {
-	if (wordByWordReplacements.length === 0) {
-		return [];
+	if (isUpperCase(name)) {
+		return {total: 0};
 	}
 
-	if (wordByWordReplacements.length === 1) {
-		return wordByWordReplacements[0];
-	}
+	const exactReplacements = getWordReplacements(name, replacements);
 
-	if (limit <= 1) {
-		return wordByWordReplacements[0];
+	if (exactReplacements.length > 0) {
+		return {
+			total: exactReplacements.length,
+			samples: exactReplacements.sort().slice(0, limit)
+		};
 	}
 
-	const [wordReplacements, ...tailWordReplacements] = wordByWordReplacements;
-	const tailCombinations = getWordByWordReplacementsCombinations(tailWordReplacements, limit / wordReplacements.length);
+	const words = name.split(/(?=[^a-z])|(?<=[^a-zA-Z])/g).filter(Boolean);
 
-	const result = [];
-	for (const name of wordReplacements) {
-		for (const combination of tailCombinations) {
-			result.push([name].concat(combination));
+	let hasReplacements = false;
+	const combined = words.map(word => {
+		const wordReplacements = getWordReplacements(word, replacements);
+		if (!hasReplacements && wordReplacements.length > 0) {
+			hasReplacements = true;
 		}
-	}
-
-	return result;
-};
 
-const getWordByWordReplacements = (replacements, normalizedName, originalIsInPascalCase) => {
-	const words = splitNormalizedName(normalizedName);
-
-	let wordByWordReplacements = words.map(word => getWordReplacements(replacements, word));
+		return wordReplacements.length > 0 ? wordReplacements : [word];
+	});
 
-	const someWordsHaveReplacements = wordByWordReplacements.some(wordReplacements => wordReplacements.length > 0);
-	if (!someWordsHaveReplacements) {
-		return [];
+	// No replacements for any word
+	if (!hasReplacements) {
+		return {total: 0};
 	}
 
-	wordByWordReplacements = wordByWordReplacements
-		.map((wordReplacements, i) => wordReplacements.length > 0 ? wordReplacements : [words[i]]);
+	const total = combined.reduce((total, {length}) => total * length, 1);
 
-	return getWordByWordReplacementsCombinations(wordByWordReplacements)
-		.map(originalIsInPascalCase ? pascalCase : camelCase)
-		.sort();
-};
-
-const getExactReplacements = (replacements, normalizedName, originalIsInPascalCase) => {
-	const variableNameReplacements = replacements.get(normalizedName);
-
-	if (!variableNameReplacements) {
-		return [];
+	if (total === 1) {
+		return {
+			total,
+			samples: [
+				combined.map(words => words[0]).join('')
+			]
+		};
 	}
 
-	return [...variableNameReplacements.keys()]
-		.filter(name => variableNameReplacements.get(name))
-		.map(originalIsInPascalCase ? pascalCase : name => name)
-		.sort();
-};
+	let options = [[]];
```

Comment:
this can be quick loop from 0 to limit, I will figure out later

Reply:
It's `cartesian product`, I've done this before but forgot how to calculate

Reply:
Why is changing `getWordByWordReplacements` even necessary here?

Reply:
I want have a function can get replacements only by word, no matter what cases, file names is more complicated than variable names